stellar-contract-env-host/src/host.rs

```diff
@@ -33,33 +36,120 @@ use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum HostError {
+    /// General error.
     #[error("general host error: {0}")]
     General(&'static str),
-    #[error("XDR error")]
-    XDRError(#[from] xdr::Error),
+    /// An error with specific status code.
+    #[error("host error: {0}, status: {:?}")]
+    WithStatus(String, ScStatus),
```

Comment:
I don't think the String is required here -- the status code should have a method that produces the symbolic name, which ought to be enough for diagnosis?